app/controllers/users_controller.rb

```diff
@@ -63,6 +64,26 @@ def destroy
     end
   end
 
+  def impersonate
+    session[:impersonated_by] = User.current.id if session[:impersonated_by].blank?
+    session[:user] = user = User.find_by_id(params[:id])
+    success _("You impersonated user %s, to cancel the session, click the eye icon in the top bar.") % user.name
+    Audit.create :auditable_type => 'User', :auditable_id => user.id, :user_id => User.current.id, :action => 'impersonate', :audited_changes => {}
```

Comment:
I wonder if we should not use `session[:impersonated_by]` instead of `User.current.id` here: in case of nested impersonation: might be harder to track the original user that did the impersonation. What do you think?

Reply:
I'm wondering if `impersonated_by` needs to be a stack or whether we should disallow nested impersonation.

Reply:
Personally I would be ok even with the way it's done now, but I can see how it could be confusing and hard to track (for example in the audit) when multiple impersonations would happen. Is there any use case where the nesting would be actually desired? If not, disabling is probably the simplest way to avoid this edge cases